src/components/PopupWithForm.js

```diff
@@ -24,11 +25,16 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", this._onPopupSubmit);
   }
 
-  _onPopupSubmit = (evt) => {
-    const event = evt;
-    this._handleFormSubmit(this._getInputValues(), event);
+  closeAfterSubmit() {
     this._popupForm.reset();
+    this._submitButton.textContent = this._submitButtonText;
     this._formValidator.toggleButtonState();
     this.close();
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`